Physical_Engine/Query/MaterialComposition.cs

```diff
@@ -71,17 +71,32 @@ public static MaterialComposition MaterialComposition(this ISurface surface)
 
         /***************************************************/
 
-        [Description("Gets all the Materials a BulkSolid is composed of and in which ratios")]
-        [Input("bulkSolids", "The BulkSolids to get the MaterialComposition from")]
-        [Output("materialComposition", "The kind of matter the BulkSolids is composed of and in which ratios", typeof(Ratio))]
-        public static MaterialComposition MaterialComposition(this BulkSolids bulkSolids)
+        [Description("Gets all the Materials a SolidBulk is composed of and in which ratios")]
+        [Input("solidBulk", "The SolidBulk to get the MaterialComposition from")]
+        [Output("materialComposition", "The kind of matter the SolidBulk is composed of and in which ratios", typeof(Ratio))]
+        public static MaterialComposition MaterialComposition(this SolidBulk solidBulk)
         {
-            if (bulkSolids.MaterialComposition == null)
+            if (solidBulk.MaterialComposition == null)
             {
-                Engine.Reflection.Compute.RecordError("The BulkSolids MaterialComposition could not be calculated as no Materials have been assigned.");
+                Engine.Reflection.Compute.RecordError("The SolidBulk MaterialComposition could not be calculated as no Materials have been assigned.");
```

Comment:
Why are you cloning the `MaterialComposition` here? Actually, do we need the method at the first place? It looks like a simple property query to me.

Reply:
Simple though it may be, `MaterialComposition` and `SolidVolume` are required extension methods for all IElementM.  😄